sys/crypto/psa_riot_cipher/chacha20.c

```diff
@@ -65,22 +66,32 @@ psa_status_t psa_cipher_chacha20_set_iv(psa_cipher_chacha20_ctx_t *ctx,
     DEBUG("RIOT ChaCha20 Cipher set IV");
     psa_status_t status = PSA_ERROR_CORRUPTION_DETECTED;
 
-    switch (iv_length)
-    {
+    /* See PSA Specification */
+    switch (iv_length) {
         case 8:
+            /* 8 bytes: the cipher operation uses the original [CHACHA20] definition
+            of ChaCha20: the provided IV is used as the 64-bit nonce, and the 64-bit
+            counter value is set to zero. */
             status = PSA_ERROR_NOT_SUPPORTED;
             break;
         case 12:
-            chacha20_setup(&ctx->ctx.ctx, ctx->ctx.key, iv, 0);
+            /* 12 bytes: the provided IV is used as the nonce, and the counter value
+            is set to zero. */
+            chacha20_setup(&ctx->ctx, ctx->buffer, iv, 0);
             status = PSA_SUCCESS;
             break;
         case 16:
-            chacha20_setup(&ctx->ctx.ctx, ctx->ctx.key, &iv[4], unaligned_get_u32(iv));
+            /* 16 bytes: the first four bytes of the IV are used as the counter value
+            (encoded as little-endian), and the remaining 12 bytes are used as the nonce. */
+            chacha20_setup(&ctx->ctx, ctx->buffer, &iv[4], unaligned_get_u32(iv));
             status = PSA_SUCCESS;
             break;
         default:
+            /* It is recommended that implementations do not support other sizes of IV. */
             status = PSA_ERROR_INVALID_ARGUMENT;
     }
+    /* Clear key from buffer */
+    memset(ctx->buffer, 0, sizeof(ctx->buffer));
```

Comment:
Maybe a place to use [`explicit_bzero`](https://doc.riot-os.org/group__sys__string__utils.html#gab3f938798fad99cd9a23c585b03f24e5) instead?